presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java

```diff
@@ -83,10 +84,11 @@ public LocationHandle forExistingTable(SemiTransactionalHiveMetastore metastore,
         }
     }
 
-    private boolean shouldUseTemporaryDirectory(HdfsContext context, Path path)
+    private boolean shouldUseTemporaryDirectory(ConnectorSession session, HdfsContext context, Path path)
     {
-        // skip using temporary directory for S3
-        return !isS3FileSystem(context, hdfsEnvironment, path);
+        return isTemporaryStagingDirectoryEnabled(session)
```

Comment:
Maybe want to update the comment around `WriteMode` enum? https://github.com/prestodb/presto/blob/79f480bc807954c4486b670c42015fb9c396f010/presto-hive/src/main/java/com/facebook/presto/hive/LocationHandle.java#L107-L121